micrometer-api/src/main/java/io/micrometer/api/instrument/docs/DocumentedSample.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.api.instrument.docs;
+
+import io.micrometer.api.instrument.Timer;
+
+/**
+ * In order to describe your spans via e.g. enums instead of Strings you can use this
+ * interface that returns all the characteristics of a span. We can
+ * analyze the sources and reuse this information to build a table of known metrics, their
+ * names, tags and events.
+ *
+ * We can generate documentation for all created spans but certain requirements need to be
+ * met.
+ *
+ * <ul>
+ *     <li>Metrics are grouped within an enum - the enum implements the {@link DocumentedSample} interface</li>
+ *     <li>If the span contains {@link TagKey} then those need to be declared as nested enums</li>
+ *     <li>The {@link DocumentedSample#getHighCardinalityTagKeys()} ()} need to call the nested enum's {@code values()} method to retrieve the array of allowed keys / events</li>
+ *     <li>The {@link DocumentedSample#getLowCardinalityTagKeys()} ()} ()} need to call the nested enum's {@code values()} method to retrieve the array of allowed keys / events</li>
+ *     <li>Javadocs around enums will be used as description</li>
+ *     <li>If you want to merge different {@link TagKey} enum {@code values()} methods you need to call the {@link TagKey#merge(TagKey[]...)} method</li>
+ * </ul>
+ *
+ * @author Marcin Grzejszczak
+ * @since 1.0.0
+ */
+public interface DocumentedSample {
+
+    /**
+     * @return metric name
+     */
+    String getName();
+
+    /**
+     * @return allowed tag keys for low cardinality tags
+     */
+    default TagKey[] getLowCardinalityTagKeys() {
+        return new TagKey[0];
```

Comment:
Since an empty array is immutable should we create a static final instance and return it instead of creating a new one?